src/app/onboarding/components/onboarding/onboarding.component.ts

```diff
@@ -74,8 +74,13 @@ export class OnboardingComponent implements OnInit {
 
   public setNewArchive(archive: ArchiveVO): void {
     this.currentArchive = archive;
-    this.account.setArchive(archive);
-    this.setScreen(OnboardingScreen.done);
+    const updateAccount = new AccountVO({defaultArchiveId: archive.archiveId});
+    this.account.updateAccount(updateAccount).then(() => {
+      this.account.setArchive(archive);
+      this.api.archive.change(archive).then(() => {
+        this.setScreen(OnboardingScreen.done);
```

Comment:
Ahh, I think I see! Let me make sure I understand the problem this is fixing:

- Archive-invited accounts have their default archive set to the pending invited archive
- Creating a new archive as part of the new onboarding flow did not update the default archive
- We were previously calling `setArchive` during onboarding, which meant that things worked for a little while - the duration of the session, basically
- But, if they logged out and back in, they were stuck with a default archive they couldn't (yet) access, and a full non-default archive prevented the onboarding from rerunning

Is that an accurate summary, @meisekimiu?

So the fix is to update the default after creating the new archive. Simple enough!

Reply:
I'm also calling `api.archive.change` here because while `setArchive` informs the `AccountService` what the current archive is, we still need to actually change to the new default archive on the back end as well (otherwise the back end still tries to fetch the pending archive).

(`account.setArchive` sets the Archive in the `AccountService` which for some reason is not automatically set by the API request)

Reply:
Oh ugh, thanks I missed that! It is always, always, always the things I think I understand but actually don't that get me; I didn't understand the distinction between `account.setArchive` and `api.archive.change`. I appreciate the explanation!